lndclient/router_client.go

```diff
@@ -0,0 +1,161 @@
+package lndclient
+
+import (
+	"context"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"google.golang.org/grpc/codes"
+
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+// RouterClient exposes payment functionality.
+type RouterClient interface {
+	SendPayment(ctx context.Context, invoice string,
+		maxFee btcutil.Amount, maxCltv *int32, outgoingChannel *uint64,
+		timeout time.Duration) (chan PaymentStatus, chan error, error)
+
+	TrackPayment(ctx context.Context, hash lntypes.Hash) (
+		chan PaymentStatus, chan error, error)
+}
+
+// PaymentStatus describe the state of a payment.
+type PaymentStatus struct {
+	State    routerrpc.PaymentState
+	Preimage lntypes.Preimage
+	Fee      lnwire.MilliSatoshi
+	Route    []uint64
+}
+
+type routerClient struct {
+	client       routerrpc.RouterClient
+	routerKitMac serializedMacaroon
+}
+
+func newRouterClient(conn *grpc.ClientConn,
+	routerKitMac serializedMacaroon) *routerClient {
+
+	return &routerClient{
+		client:       routerrpc.NewRouterClient(conn),
+		routerKitMac: routerKitMac,
+	}
+}
+
+func (r *routerClient) SendPayment(ctx context.Context, invoice string,
+	maxFee btcutil.Amount, maxCltv *int32, outgoingChannel *uint64,
+	timeout time.Duration) (chan PaymentStatus, chan error, error) {
```

Comment:
If we have the structs like `TrackPaymentMessage` below, why not use them here instead of unrolling all the fields into individual arguments?

Reply:
Can't use `TrackPaymentMessage` because it is in the test package, but created a new struct for this.